src/util/Logging.h

```diff
@@ -13,22 +13,39 @@
 #define ELPP_WINSOCK2
 #define ELPP_DEBUG_ERRORS
 
-// NOTE: Nothing else should include easylogging directly include this file instead
-// Please think carefully modifying this file, and potentially using synchronization
-// primitives. It is easy to introduce data races and deadlocks, so it is recommended
-// to use valgrind --tool=helgrind to detect potential problems.
+// NOTE: Nothing else should include easylogging directly include this file
+// instead Please think carefully modifying this file, and potentially using
+// synchronization primitives. It is easy to introduce data races and deadlocks,
+// so it is recommended to use valgrind --tool=helgrind to detect potential
+// problems.
 #include "lib/util/easylogging++.h"
 
 namespace stellar
 {
 class Logging
 {
     static el::Configurations gDefaultConf;
+
+    template <typename T> struct LockElObject
+    {
+        T* const mItem;
+        LockElObject(T* elObj);
+        ~LockElObject();
+
+        // Disable copy construction and copy assignment
+        LockElObject(LockElObject const&) = delete;
+        LockElObject& operator=(LockElObject const&) = delete;
+    };
+
     struct LockHelper
```

Comment:
`LockElObject` and `LockHelper` are only used in the `.cpp` file so they should be moved there. Cleaner and will make compilation faster if future maintainers ever change them.